source/RoslynAnalyzers/Descriptors.cs

```diff
@@ -86,5 +86,151 @@ public static DiagnosticDescriptor Oct2008IntegrationTestForwardCancellationToke
                 "Forward the CancellationToken property to compatible methods to ensure cancellation notifications get properly propagated,"
                 + "or pass in 'CancellationToken.None' explicitly to indicate intentionally not propagating the token.");
         }
+        
+        // ----- Message Contract Analyzers -----
+        
+        const string Category = "Octopus";
+
+        public static readonly DiagnosticDescriptor EventTypesMustBeNamedCorrectly = new(
+            "OCT3001",
+            "Event types must either end with Event or EventV[versionNumber]",
+            "Event types must either end with Event or EventV[versionNumber]",
+            Category,
+            DiagnosticSeverity.Error,
+            true);
+        
+        public static readonly DiagnosticDescriptor CommandTypesMustBeNamedCorrectly = new(
+            "OCT3002",
+            "Command types must either end with Command or CommandV[versionNumber]",
+            "Command types must either end with Command or CommandV[versionNumber]",
+            Category,
+            DiagnosticSeverity.Error,
+            true);
+
+        public static readonly DiagnosticDescriptor RequestTypesMustBeNamedCorrectly = new(
+            "OCT3003",
+            "Request types must either end with Request or RequestV[versionNumber]",
+            "Request types must either end with Request or RequestV[versionNumber]",
+            Category,
+            DiagnosticSeverity.Error,
+            true);
+
+        public static readonly DiagnosticDescriptor CommandTypesMustHaveCorrectlyNamedResponseTypes = new(
+            "OCT3004",
+            "Types that implement ICommand must have responses with matching names",
+            "Response type should be \"{0}\" instead of \"{1}\" (Types that implement ICommand must have responses with matching names)",
+            Category,
+            DiagnosticSeverity.Error,
+            true);
+
+        public static readonly DiagnosticDescriptor RequestTypesMustHaveCorrectlyNamedResponseTypes = new(
+            "OCT3005",
+            "Types that implement IRequest must have responses with matching names",
+            "Response type should be \"{0}\" instead of \"{1}\" (Types that implement IRequest must have responses with matching names)",
```

Comment:
```suggestion
            "Response type should be \"{0}\" instead of \"{1}\" (types that implement IRequest must have responses with matching names)",
```